src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -254,6 +268,25 @@ public void CancelRun(bool force)
         {
             foreach (var shift in Shifts)
                 shift.Cancel(force);
+
+            if (force)
+            {
+                SpinWait.SpinUntil(() => _topLevelWorkItem.State == WorkItemState.Complete, WAIT_FOR_FORCED_TERMINATION);
+
+                // Notify termination of any remaining in-process suites
+                lock (_activeWorkItems)
+                {
+                    int index = _activeWorkItems.Count;
+
+                    while (index > 0)
+                    {
+                        var work = _activeWorkItems[--index];
```

Comment:
(Not important) Moving the decrement to a `for` loop is easier for me personally to parse than using the value of a decrement expression.

Reply:
Definitely stylistic, I think. My preference comes from having started writing pdp-11 assembly code, where auto-decrementing registers are a thing. Also, very popular in K&R C. :-)